house/tests/house_tests.py

```diff
@@ -58,3 +61,58 @@ def test_delete_house(self, generate_house):
         generate_house.save()
         generate_house.delete()
         assert generate_house not in House.objects.all()
+
+    def test_filter_houses_by_form(self):
+        form_data = {
+            'country': Country.objects.get(name='Israel'),
+            'city': City.objects.get(name='Tel Aviv'),
+            'parent_profession_1': HOUSE_PARENT_PROFESSION_1,
+            'parent_profession_2': None,
+            'highest_income': 10000,
+            'lowest_income': 5000,
+            'children': 1,
+        }
+        # test parent_profession_1 filter
+        pass_filter_house_create()
+        HouseFactory(country=Country.objects.get(name='Israel'), city=City.objects.get(name='Tel Aviv'), income=5000,
+                     children=1, parent_profession_1=Job.OTHER).save()
+        houses = _filter_houses_by_form(form_data, House.objects.all())
+        assert len(houses) == 1
+        House.objects.all().delete()
+
+        # test country filter
+        pass_filter_house_create()
+        HouseFactory(country=Country.objects.get(name='India'), city=City.objects.get(name='Tel Aviv'), income=5000,
+                     children=1, parent_profession_1=Job.TEACHER).save()
+        houses = _filter_houses_by_form(form_data, House.objects.all())
+        assert len(houses) == 1
+        House.objects.all().delete()
+
+        # test city filter
+        pass_filter_house_create()
+        HouseFactory(country=Country.objects.get(name='Israel'), city=City.objects.get(name='Ramat Gan'), income=5000,
+                     children=1, parent_profession_1=Job.TEACHER).save()
+        houses = _filter_houses_by_form(form_data, House.objects.all())
+        assert len(houses) == 1
+        House.objects.all().delete()
+
+        # test income filter (highest_income)
+        pass_filter_house_create()
+        HouseFactory(country=Country.objects.get(name='Israel'), city=City.objects.get(name='Tel Aviv'), income=12000,
+                     children=1, parent_profession_1=Job.TEACHER).save()
+        houses = _filter_houses_by_form(form_data, House.objects.all())
+        assert len(houses) == 1
+        House.objects.all().delete()
+
+        # test income filter (lowest_income)
+        pass_filter_house_create()
+        HouseFactory(country=Country.objects.get(name='Israel'), city=City.objects.get(name='Tel Aviv'), income=4000,
+                     children=1, parent_profession_1=Job.TEACHER).save()
+        houses = _filter_houses_by_form(form_data, House.objects.all())
+        assert len(houses) == 1
+        House.objects.all().delete()
+
```

Comment:
Can you split the test into small functions so each function is a test on its own? 
It is not clear at first read what each test is for. 
As you wrote good code is self-explanatory and comments are not supposed to be critical to the understanding of the reviewer/contributor.